### PR TITLE
Digitizer mch  mclabels  & MCH digitizer testing

### PR DESCRIPTION
**Pull request by mwinn2**
- implementation of MC labels for MCH
- implementation of digit merging for digits and MC labels
- testing functionality for digitizer, response and algorithm speed comparison for different mergings

### CODE REVIEW

**Comment by sawenzel** [comment]
This should probably be squashed when approved.

**Comment by aphecetche** [comment]
@sawenzel indeed, squashed it should be. The o2checkcode error I don't understand though.

@mwinn2 could you please have a look and try to pacify clang-format (Travis failure) and Codacy ?

**Comment by mwinn2** [comment]
@sawenzel , @aphecetche Ok, the code checker should be fine now, the clang part, I am not sure how to fix,  clang-format -style=file -i ... is fine for the .h and .cxx as far as I can see and for lists and linkdef is doesn't compile when applied blindly. I will have later today (US west coast time)

**Comment by mwinn2** [comment]
@sawenzel , @aphecetche, ok, I checked the cxx and .h should be fine as they are.

File: Steer/DigitizerWorkflow/src/MCHDigitizerSpec.cxx [hunk 1]
```diff
@@ -111,15 +110,16 @@ class MCHDPLDigitizerTask
         retrieveHits(mSimChains, "MCHHit", part.sourceID, part.entryID, &hits);
         LOG(INFO) << "For collision " << collID << " eventID " << part.entryID << " found MCH " << hits.size() << " hits ";
```
Comment (aphecetche) [inline Steer/DigitizerWorkflow/src/MCHDigitizerSpec.cxx]: I guess all those LOG(INFO) should be removed (or at least be demoted to LOG(DEBUG)) ?

Comment (mwinn2) [reply Steer/DigitizerWorkflow/src/MCHDigitizerSpec.cxx]: @aphecetche : not clear to me, for the other detectors, I see quite extensive log output. @sawenzel : what do you think?

**Comment by mwinn2** [comment]
@aphecetche @sawenzel : the problem with the clang-tidy must reside in cmake/O2Dependencies.cmake, Detectors/MUON/MCH/Simulation/test/CMakeLists.txt or in something that is not automatically caught by clang-format -style=file -i FILE

File: Steer/DigitizerWorkflow/src/MCHDigitizerSpec.cxx (hunk 1 above)
Comment (sawenzel) [reply Steer/DigitizerWorkflow/src/MCHDigitizerSpec.cxx]: True that we have this in a couple of places, but I guess we can start to reduce this as it was primarily meant for debugging at the beginning.

Comment (mwinn2) [reply Steer/DigitizerWorkflow/src/MCHDigitizerSpec.cxx]: Ok, I replaced now by LOG(DEBUG).

**Pull request merged**